autogtp/main.cpp

```diff
@@ -50,26 +51,27 @@ bool fetch_best_network_hash(QTextStream& cerr, QString& nethash) {
         exit(EXIT_FAILURE);
     }
     QString outhash = outlst[0];
-    cerr << "Best network hash: " << outhash << endl;
     QString client_version = outlst[1];
     auto server_expected = client_version.toInt();
-    cerr << "Required client version: " << client_version;
     if (server_expected > AUTOGTP_VERSION) {
-        cerr << endl;
         cerr << "Server requires client version " << server_expected
              << " but we are version " << AUTOGTP_VERSION << endl;
         cerr << "Check https://github.com/gcp/leela-zero for updates." << endl;
         exit(EXIT_FAILURE);
-    } else {
-        cerr << " (OK)" << endl;
+    }
+    if (games_played == 0) {
+        cerr << "Best network hash: " << outhash << endl;
+        cerr << "Required client version: " << server_expected << " (OK)" << endl;
     }
     nethash = outhash;
     return true;
 }
 
-bool fetch_best_network(QTextStream& cerr, QString& netname) {
+bool fetch_best_network(QTextStream& cerr, QString& netname, long games_played) {
     if (QFileInfo::exists(netname)) {
-        cerr << "Already downloaded network." << endl;
+        if (games_played == 0) {
+            cerr << "Already downloaded network." << endl;
```

Comment:
Same remark, there's no reason why this would only apply to the first game.

Reply:
If the best network updates then this will be skipped (because the file won't exist)and the logging at line 108 will print showing an update.